exporter/datadogexporter/config.go

```diff
@@ -498,6 +396,76 @@ func (c *Config) Validate() error {
 	return nil
 }
 
+var _ error = (*renameError)(nil)
+
+// renameError is an error related to a renamed setting.
+type renameError struct {
+	// oldName of the configuration option.
+	oldName string
+	// newName of the configuration option.
+	newName string
+	// issueNumber on opentelemetry-collector-contrib for tracking
+	issueNumber uint
+}
+
+// List of settings that have been removed, but for which we keep a custom error.
+var removedSettings = []renameError{
+	{
+		oldName:     "metrics::send_monotonic_counter",
+		newName:     "metrics::sums::cumulative_monotonic_mode",
+		issueNumber: 8489,
+	},
+	{
+		oldName:     "tags",
+		newName:     "host_metadata::tags",
+		issueNumber: 9099,
+	},
+	{
+		oldName:     "send_metadata",
+		newName:     "host_metadata::enabled",
+		issueNumber: 9099,
+	},
+	{
+		oldName:     "use_resource_metadata",
+		newName:     "host_metadata::hostname_source",
+		issueNumber: 9099,
+	},
+	{
+		oldName:     "metrics::report_quantiles",
+		newName:     "metrics::summaries::mode",
+		issueNumber: 8845,
+	},
+	{
+		oldName:     "metrics::instrumentation_library_metadata_as_tags",
+		newName:     "metrics::instrumentation_scope_as_tags",
+		issueNumber: 11135,
+	},
+}
+
+// Error implements the error interface.
+func (e renameError) Error() string {
+	return fmt.Sprintf(
+		"%q was removed in favor of %q. See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/%d",
+		e.oldName,
+		e.newName,
+		e.issueNumber,
+	)
+}
+
+// OccursIn reports whether the removed config setting defined in e is used
+func (e renameError) OccursIn(configMap *confmap.Conf) bool {
```

Comment:
I'd argue now that this isn't necessary anymore, since you can just call `configMap.IsSet` below inside `handleRemovedSettings` 😓

Reply:
Yes, makes sense, removed